src/createServer.js

```diff
@@ -1,8 +1,97 @@
 'use strict';
 
+const http = require('http');
+const fs = require('fs');
+const zlib = require('zlib');
+const formidable = require('formidable');
+const { pipeline } = require('stream');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = new http.Server();
+
+  server.on('request', (req, res) => {
+    if (req.url === '/' && req.method === 'GET') {
+      res.statusCode = 200;
+      res.setHeader('Content-Type', 'text/html');
+      fs.createReadStream('./public/index.html').pipe(res);
+
+      return;
+    }
+
+    if (req.url === '/compress' && req.method === 'GET') {
+      res.statusCode = 400;
+      res.setHeader('Content-Type', 'text/plain');
+
+      res.end('Requests to /compress need to be POST method');
+
+      return;
+    }
+
+    if (req.url === '/compress' && req.method === 'POST') {
+      const form = new formidable.IncomingForm({
+        filename: (name, ext, part) => part.originalFilename,
+      });
+
+      form.parse(req, (err, { compressionType }, { file }) => {
+        if (!file || !compressionType || err) {
+          res.statusCode = 400;
+          res.end('Form error');
+
+          return;
+        }
+
+        if (!['gzip', 'br', 'deflate'].includes(compressionType[0])) {
+          res.statusCode = 400;
+          res.end('Unsupported compression type');
+
+          return;
+        }
+
+        const fileStream = fs.createReadStream(file[0].filepath);
+
+        res.statusCode = 200;
+
+        res.setHeader(
+          'Content-Disposition',
+          `attachment; filename=${file[0].originalFilename}.${compressionType[0]}`,
+        );
+
+        let compressorType;
```

Comment:
Not sure, but you could try:
declaring your previous Object literal here, might work just like this, but better:

```
const compressors = {
  gzip: zlib.createGzip,
  br: zlib.createBrotliCompress,
  deflate: zlib.createDeflate,
};

const compressorType = compressors[compressionType[0]]();
```

See how I did not call the functions inside the `compressors` object, but only in `compressorType`, after I already know which will be used.

And, if you store this `compressors` object before line 43, you can even replace `!['gzip, 'br', 'deflate'].includes...` for `!Object.keys(compressors).includes...`
Allowing your coed to be easily extendable later on(you would only need to change the `compressors` object)